services/src/test/java/org/apache/druid/testing/embedded/EmbeddedClusterApis.java

```diff
@@ -116,6 +116,20 @@ public void waitForTaskToSucceed(String taskId, EmbeddedOverlord overlord)
     verifyTaskHasStatus(taskId, TaskStatus.success(taskId));
   }
 
+  /**
+   * Waits for the given task to fail, with the given errorMsq. If the given
+   * {@link EmbeddedOverlord} is not the leader, this method can only return by
+   * throwing an exception upon timeout.
+   */
+  public void waitForTaskToFail(String taskId, String errorMsg, EmbeddedOverlord overlord)
```

Comment:
Better for this to take a `Matcher<String>` as the last argument, so it would be called like:

```java
apis.waitForTaskToFail(taskId, overlord, CoreMatchers.equalTo("xyz"));
```

That way any kind of matching can be done, not just equality.

Reply:
I am good with `waitForTaskToFinish`, that's even simpler and more generally useful. It should return a task status object and not do any assertions on it. The `waitForTaskToSucceed` method should be rewritten to use `waitForTaskToFinish`.